paperoni/papers2.py

```diff
@@ -137,3 +138,10 @@ def get_ref(self, link_type):
                 return link.ref
         else:
             return None
+
+    def download_pdf(self):
+        for link in self.links:
+            if link.type == "pdf":
+                print(link.ref)
```

Comment:
This should actually download the PDF. There is already code to do this (uses the request module and tqdm for the progress bar, if I recall correctly).